ci/scripts/cpp_build.sh

```diff
@@ -118,12 +118,20 @@ if [ "${ARROW_USE_MESON:-OFF}" = "ON" ]; then
     fi
   }
 
+  ORIGINAL_CXX=${CXX}
+  if [ "${ARROW_USE_CCACHE}" = "ON" ]; then
+      export CXX="ccache ${CXX}"
+  else
+      export CXX="sccache ${CXX}"
+  fi
   meson setup \
     --prefix=${MESON_PREFIX:-${ARROW_HOME}} \
     --buildtype=${ARROW_BUILD_TYPE:-debug} \
     -Dtests=$(meson_boolean ${ARROW_BUILD_TESTS:-OFF}) \
     . \
     ${source_dir}
+
+  export CXX=${ORIGINAL_CXX}
```

Comment:
I think that we don't need `export` here:

```suggestion
  CXX="${ORIGINAL_CXX}"
```

If `CXX` is already exported, we don't need to `export` here. If `CXX` isn't exported, we don't need to export an empty `CXX`.